tests/test_scenarios.py

```diff
@@ -37,53 +48,119 @@ def test_scenario(filename, monkeypatch):
             subprocess.check_call(["git", "init"])
             subprocess.check_call(["git", "add", "-N", "."])
             subprocess.check_call(["git", "commit", "-a", "-m", "foo"])
-        # TODO handle options
-        result = runner.invoke(main, command, catch_exceptions=False)
-
-    cleaned_output = LOG_LINE_TIMESTAMP_RE.sub("", result.output)
-    cleaned_output = LOG_LINE_NUMERIC_LINE_RE.sub(lambda m: (m.group(1) + "<n>"), cleaned_output)
-    cleaned_output = GIT_VERSION_RE.sub(lambda m: (m.group(1) + "<stuff>"), cleaned_output)
-    cleaned_output = re.sub(r"(?m) +$", "", cleaned_output)
-
-    if os.getenv("UPDATE_SCENARIOS"):
-        save_scenario(path, cleaned_output)
-    else:
-        assert output == cleaned_output
+        for command in commands:
+            if command.command[:6] == "$ ick ":
+                # TODO: handle global options like -vv
+                args = shlex.split(command.command[6:])
+                result = runner.invoke(main, args, catch_exceptions=False)
+                output = result.output
+            elif command.command == "":
+                output = ""
+            else:
+                assert command.command[:2] == "$ "
+                proc = subprocess.run(
+                    command.command[2:],
+                    encoding="utf-8",
+                    shell=True,
```

Comment:
Could you include a couple of lines about why shell (I think it's for handling redirection and also semicolons)